src/legacy/core_plugins/data/public/filter/filter_manager/filter_manager.ts

```diff
@@ -178,8 +182,13 @@ export class FilterManager {
     }
   }
 
-  public invertFilter(filter: Filter) {
-    filter.meta.negate = !filter.meta.negate;
+  public async invertFilter(filter: Filter) {
+    const newFilter = _.cloneDeep(filter);
```

Comment:
hmm, seems before this method only inverted filter, but didn't add it to filter manager ?

this method now does both (creates new filter, inverts it, adds it to the filter manager) and also modifies the existing filter.

imo it should do one or the other (preferably just invert existing filter)

Reply:
Chaneged implementation completely

Reply:
I actually think the newest implementation is confusing because reading it, I would think that `filterManager.invertFilter(f1)` would do the updating because you are calling it on an instance and it isn't returning anything.

What do you think about just having a helper method that is not on the filterManager instance at all, then you can do something like

```ts
import { invertFilter } from '../invert_filter';

test('...', () => {
     const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
     const invertedFilter = invertFilter(f1);
     await filterManager.addFilters(invertedFilter);
     const newFilters1 = filterManager.getFilters()[0];
     expect(invertedFilter.meta.negate).toBe(true);
     expect(newFilters1.meta.negate).toBe(true);
}
```

By returning a new filter object, we aren't encouraging manipulating the filter objects directly, and the user should have a clearer indication they then need to add it.

It's interesting because tests for what happens when adding an inverted filter already exist they just don't use that `invertFilter` helper method (see `should allow overwriting a positive filter by a negated one`).